### PR TITLE
Celery worker doesn't show log on UI

### PR DESCRIPTION
**Pull request by hxquangnhat**
Airflow Celery worker doesn't show log
- add new `worker-svc.yaml` in templates folder for celery headless service
- modify `worker-sts.yaml` with new headless service
- modify worker sts in `airflow.go`

### CODE REVIEW

File: pkg/apis/airflow/v1alpha1/airflow.go [hunk 1]
```diff
@@ -339,6 +339,10 @@ func tmplsvc(v interface{}) (*resource.Item, error) {
 	return k8s.ItemFromFile(TemplatePath+"svc.yaml", v, &corev1.ServiceList{})
 }
 
+func tmpWorkerSvc(v interface{}) (*resource.Item, error) {
```
Comment (barney-s) [inline pkg/apis/airflow/v1alpha1/airflow.go]: could you rename it to tmpl

File: templates/worker-svc.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
```
Comment (barney-s) [inline templates/worker-svc.yaml]: This seems identical to https://github.com/GoogleCloudPlatform/airflow-operator/blob/master/templates/svc.yaml
Do we need a new file ?

File: pkg/apis/airflow/v1alpha1/airflow.go (hunk 1 above)
Comment (barney-s) [reply pkg/apis/airflow/v1alpha1/airflow.go]: This function may be redundant. See if you can use tmplsvc() itself.

File: pkg/apis/airflow/v1alpha1/airflow.go [hunk 3]
```diff
@@ -947,14 +951,15 @@ func (s *WorkerSpec) ExpectedResources(rsrc interface{}, rsrclabels map[string]s
...
+		SvcName:    rsrcName(r.Name, ValueAirflowComponentWorker, ""),
 		Cluster:    r,
 		Base:       base,
 		Labels:     rsrclabels,
 		Selector:   rsrclabels,
 		Ports:      map[string]string{"wlog": "8793"},
 	}
 
-	for _, fn := range []resource.GetItemFn{s.sts} {
+	for _, fn := range []resource.GetItemFn{tmpWorkerSvc, s.sts} {
```
Comment (barney-s) [inline pkg/apis/airflow/v1alpha1/airflow.go]: See if you can use tmplsvc() here instead.

File: templates/worker-svc.yaml (hunk 2 above)
Comment (hxquangnhat) [reply templates/worker-svc.yaml]: The difference is there's a field `clusterIP: None` for creating the headless service.

**Comment by hxquangnhat** [comment]
I'll recheck if we can use the `tmplsvc()`, as the worker service is a headless one.

File: templates/worker-svc.yaml (hunk 2 above)
Comment (barney-s) [reply templates/worker-svc.yaml]: Ah i see. Then rename the file to headlesssvc.yaml and create tmplHeadlessSvc().

You do need 2 files.

**Comment by barney-s** [comment]
> I'll recheck if we can use the `tmplsvc()`, as the worker service is a headless one.

You would need 2 files. create new tmplHeadlessSvc() and rename the template to headlesssvc.yaml

**Comment by hxquangnhat** [comment]
Thanks for the suggestion, it's done.

**Pull request merged**